WallsAndHoles/mapview.cpp

```diff
@@ -77,6 +76,33 @@ void MapView::createMap(TileMap *tileMap)
     }
 }
 
+void MapView::genHeightMap(TileMap *tileMap){
```

Comment:
This should all probably be handled by the MapCell itself: something in the form of
`for (...) mMapCells(x, y).setMode(Height);`

It is more or less by design that MapView doesn't have a reference to its TileMap, but each MapCell has a reference to its cell.